adapters/mqtt-vertx/src/main/java/org/eclipse/hono/adapter/mqtt/Application.java

```diff
@@ -0,0 +1,162 @@
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * The Hono MQTT adapter main application class.
+ */
+@ComponentScan
+@Configuration
+@EnableAutoConfiguration
+public class Application {
+
+    private static final Logger LOG = LoggerFactory.getLogger(Application.class);
+
+    @Autowired
+    private Vertx vertx;
+    @Value(value = "${hono.maxinstances:1}")
+    private int maxInstances;
+    @Value(value = "${hono.startuptimeout:20}")
+    private int startupTimeout;
+    @Autowired
+    private MqttAdapterFactory factory;
+    private AtomicBoolean running = new AtomicBoolean();
+
+    @PostConstruct
+    public void registerVerticles() {
+
+        if (running.compareAndSet(false, true)) {
+            final int instanceCount;
+            if (this.maxInstances > 0 && this.maxInstances < Runtime.getRuntime().availableProcessors()) {
+                instanceCount = this.maxInstances;
+            } else {
+                instanceCount = Runtime.getRuntime().availableProcessors();
+            }
+
+            try {
+
+                final CountDownLatch latch = new CountDownLatch(1);
+                final Future<Void> startFuture = Future.future();
+                startFuture.setHandler(done -> {
+                    if (done.succeeded()) {
+                        latch.countDown();
+                    } else {
+                        LOG.error("could not start MQTT adapter", done.cause());
+                    }
+                });
+
+                this.deployVerticle(instanceCount, startFuture);
+
+                if (latch.await(this.startupTimeout, TimeUnit.SECONDS)) {
+                    LOG.info("MQTT adapter startup completed successfully");
+                } else {
+                    LOG.error("startup timed out after {} seconds, shutting down ...", this.startupTimeout);
+                    this.shutdown();
+                }
+
+            } catch (InterruptedException e) {
+                LOG.error("startup process has been interrupted, shutting down ...");
+                Thread.currentThread().interrupt();
+                this.shutdown();
+            }
+        }
+    }
+
+    private void deployVerticle(int instanceCount, Future<Void> resultHandler) {
```

Comment:
we should check whether we have configured this in the eclipse codestyle rules as well..